alibuild_helpers/build_template.sh

```diff
@@ -84,6 +84,24 @@ if [[ $DEVEL_PREFIX ]]; then
   ln -snf $PKGHASH $WORK_DIR/BUILD/$PKGNAME-latest-$DEVEL_PREFIX
 fi
 
+# Create etc/profile/init.sh early, to help with debugging failed builds.
+# If we're unpacking a tarball, instead of building a package, the init.sh
+# from the tarball will replace the one we create here, but that should be
+# safe.
+mkdir -p "$INSTALLROOT/etc/profile.d"
+cat <<\EOF > "$INSTALLROOT/etc/profile.d/init.sh"
+# Source dependencies' init.sh.
+%(source_dependencies)s
+# Set up variables pointing to this package.
+%(package_vars)s
+# Environment variables set by this alidist recipe.
+%(environment)s
```

Comment:
This might actually be the reason why we did not do this before. Environment variables created by a package should be effective only after the build. Does that change the behaviour?

Reply:
No, it doesn't -- this block is inside the heredoc, so it only creates `init.sh`. It doesn't `source` it. Further up, only `%(source_dependencies)s` is executed.

The problem is that the order used to be "unpack tarball" -> "overwrite init.sh" (so the newly-generated init.sh wins); this changes it to "create init.sh" -> "unpack tarball, including init.sh, over the top" (i.e. tarball wins). But as I say in the comment, the files ought to be the same anyway I think...